lib/motsu/src/context.rs

```diff
@@ -1,34 +1,98 @@
 //! Unit-testing context for Stylus contracts.
-use std::sync::{Mutex, MutexGuard};
 
+use std::{collections::HashMap, ptr};
+
+use dashmap::DashMap;
+use once_cell::sync::Lazy;
 use stylus_sdk::{alloy_primitives::uint, prelude::StorageType};
 
-use crate::storage::reset_storage;
+use crate::prelude::{Bytes32, WORD_BYTES};
 
-/// A global static mutex.
-///
-/// We use this for scenarios where concurrent mutation of storage is wanted.
-/// For example, when a test harness is running, this ensures each test
-/// accesses storage in an non-overlapping manner.
+/// Context of stylus unit tests associated with the current test thread.
+#[allow(clippy::module_name_repetitions)]
+pub struct TestContext {
```

Comment:
I believe we should it rename it just to `Context`. It is in `motsu` library so it should be obvious.

Reply:
It is mostly a question whether `Context::current().set_msg_sender("0x15a3...")` or `TestContext::current().set_msg_sender("0x15a3...")` better.
I probably agree with you. Short version still understandable.
Curious what @ggonzalez94 think?